server/weaverbird/pipeline_executor.py

```diff
@@ -26,12 +30,17 @@ def execute_pipeline(self, pipeline_steps: List[dict]) -> DataFrame:
         # self.validate_pipeline()
         steps = Pipeline(steps=pipeline_steps).steps
         df = None
+        stopwatch = StopWatch()
         for index, step in enumerate(steps):
             try:
-                df = step.execute(
-                    df,
-                    domain_retriever=self.retrieve_domain,
-                    execute_pipeline=self.execute_pipeline,
+                with stopwatch:
+                    df = step.execute(
+                        df,
+                        domain_retriever=self.retrieve_domain,
+                        execute_pipeline=self.execute_pipeline,
+                    )
+                logger.info(
+                    f'step {step} used {convert_size(df.memory_usage().sum())}, took {stopwatch.interval}s to execute'
```

Comment:
This makes me think of something else :
to be sure that the memory usage stays close to the memory of the current dataframes, maybe it'll be nice to ensure previous dataframes are freed from the memory.

So maybe we should call the garbage collector explicitly after each step?
https://stackoverflow.com/questions/1316767/how-can-i-explicitly-free-memory-in-python